src/base/bittorrent/session.cpp

```diff
@@ -1820,14 +1820,26 @@ void Session::configureNetworkInterfaces(lt::settings_pack &settingsPack)
             {
                 endpoints << (guid + portString);
                 outgoingInterfaces << guid;
+                // Cache the GUID since on next startup the interface/VPN may be disconnected and GUID will be unavailable.
+                SettingsStorage::instance()->storeValue(u"Network/InterfaceGUID"_qs, guid);
             }
             else
             {
-                LogMsg(tr("Could not find GUID of network interface. Interface: \"%1\"").arg(ip), Log::WARNING);
-                // Since we can't get the GUID, we'll pass the interface name instead.
-                // Otherwise an empty string will be passed to outgoing_interface which will cause IP leak.
-                endpoints << (ip + portString);
-                outgoingInterfaces << ip;
+                // Since we can't get the GUID, we'll use the cached GUID if available. Otherwise we'll pass the interface name.
+                // Passing an empty string will cause IP leak so we avoid that.
+                const QString value = SettingsStorage::instance()->loadValue<QString>(u"Network/InterfaceGUID"_qs);
```

Comment:
Is this code correct? What if it loops more than one time into this code? Won't you get duplicates in the final result?

I feel you should reevaluate it. Doing work in this function doesn't look right to me.